test/truefi2/AaveBaseRateOracle.test.ts

```diff
@@ -0,0 +1,268 @@
+          .to.be.revertedWith('AaveBaseRateOracle: Number of values should be greater than 1')
+      })
+
+      it('numberOfValues is greater than buffer size', async () => {
+        const numberOfValues = BUFFER_SIZE + 1
+        await expect(oracleShortCooldown.calculateAverageAPY(numberOfValues))
+          .to.be.revertedWith('AaveBaseRateOracle: Number of values is limited by buffer size')
+      })
+    })
+
+    describe('calculates the rate correctly', () => {
+      it('before any update call', async () => {
+        await expect(aaveBaseRateOracle.calculateAverageAPY(2))
+          .to.be.revertedWith('AaveBaseRateOracle: There are fewer values stored than required')
+      })
+
+      it('with partially overwritten buffer', async () => {
+        await mockAaveLendingPool.mock.getReserveData.returns(...mockReserveData(BN(1e27)))
+        await updateBufferRightAfterCooldown(aaveBaseRateOracle)
+        await mockAaveLendingPool.mock.getReserveData.returns(...mockReserveData(BN(2e27)))
+        await updateBufferRightAfterCooldown(aaveBaseRateOracle)
+        // Aave variable borrow apys (percents): 1.0, 2.0 probed with 1 day interval
+        // Expected value is 3/2 = 1.5
+        expect(await aaveBaseRateOracle.calculateAverageAPY(3)).to.eq(1_5000)
+      })
+
+      it('with overwritten buffer', async () => {
+        await mockAaveLendingPool.mock.getReserveData.returns(...mockReserveData(BN(2e27)))
+        for (let i = 0; i < BUFFER_SIZE; i++) {
+          await updateBufferRightAfterCooldown(aaveBaseRateOracle)
+        }
+        await mockAaveLendingPool.mock.getReserveData.returns(...mockReserveData(BN(3e27)))
+        await updateBufferRightAfterCooldown(aaveBaseRateOracle)
+        // Aave variable borrow apys (percents): 2.0, 2.0, 2.0, 2.0, 2.0, 2.0, 3.0 probed with 1 day interval
+        // Expected value is 13 / 6 = 2.1(6)
+        expect(await aaveBaseRateOracle.calculateAverageAPY(7)).to.eq(2_1666)
+      })
+
+      it('spot apy value has no impact on average apy', async () => {
+        await mockAaveLendingPool.mock.getReserveData.returns(...mockReserveData(BN(1e27)))
+        await updateBufferRightAfterCooldown(aaveBaseRateOracle)
+        await mockAaveLendingPool.mock.getReserveData.returns(...mockReserveData(BN(2e27)))
+        await updateBufferRightAfterCooldown(aaveBaseRateOracle)
+        await mockAaveLendingPool.mock.getReserveData.returns(...mockReserveData(BN(3e27)))
+        await updateBufferRightAfterCooldown(aaveBaseRateOracle)
+        // Aave variable borrow apys (percents): 1.0, 2.0, 3.0 probed with 1 day interval
+        // Expected value is (1.0 + 2.0 + 3.0) / 3 = 2.0
+        expect(await aaveBaseRateOracle.calculateAverageAPY(4)).to.be.closeTo(BN(2_0000), 1)
+      })
+    })
+
+    describe('getWeeklyAPY', () => {
+      describe('returns correct value if', () => {
+        it('prices grows for last 3 days', async () => {
```

Comment:
Please fix weekly tests to calculate over 7 days instead of 6. Setting outlier values on the edge cases (e.g., 100 or 10000 for day 0 or 1) would help catch off by one errors.